### ISSUE TITLE
Consider Imageio for image loading

### ISSUE THREAD

**Issue by freeman-lab** [freeman-lab/thunder#85]
We have had repeated issues with PIL / pillow, including malfunctioning installations on certain OS X versions, as well as tif-loading behavior that works in pillow but not PIL (see e.g. #82 ).

@broxtronix suggested we check out https://github.com/imageio/imageio. It looks promising. We should evaluate whether it has sufficient functionality to replace all of our current usages of PIL / pillow.

**Comment by broxtronix** [comment]
Yes, I think imageio is definitely worth a look.  Their strategy seems to be to download the necessary native libraries for your platform when they are installed / used.  That's tricky to get right, but from what I've seen they have done a good job at working out those fiddly details.  I don't think they have great support yet for tiff stacks and other 3D datatypes (although some support is there), but my colleague Anselm has had good interactions getting code merged into their project.  I think they would be receptive to any improvement necessary to get it to work harmoniously with Thunder!

One other question I don't know the answer to is whether imageio can support encoding / decoding of compressed image data in RAM (as opposed to files on disk) as PIL/Pillow does.  I know you guys rely on that functionality!

The other option (which we have used here at Stanford) is to create our own reading / writing routines that use various different libraries (libtiff, OpenCV, scipy.io) as needed for different file formats.  This is a "roll your own" sort of approach that gives the most control.  However, I wouldn't particularly recommend it since it means dealing with all the platform-specific details yourself!
